src/megaclient.cpp

```diff
@@ -909,10 +909,18 @@ void MegaClient::exec()
 
                         // notify app in case some attributes were not returned, then redispatch
                         fc->failed(this);
+                        fc->req.disconnect();
+                        fc->req.status = REQ_PREPARED;
+                        fc->timeout.reset();
                         fc->bt.reset();
                         break;
 
                     case REQ_INFLIGHT:
+                        if (!fc->req.httpio)
+                        {
+                            break;
+                        }
+
```

Comment:
Don't run the code related to this state if the request is not really active. The state `REQ_INFLIGHT` is also set for this request when the download URL is being fetched from API servers (using `pendingcs`) and, during that time, nor the code below nor the timeout should be active.